### PR TITLE
[Merged by Bors] - feat(Data/Matrix/Basic): 0 ≤ 1

### PR DESCRIPTION
**Pull request by madvorak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Matrix/Basic.lean [hunk 1]
```diff
@@ -567,6 +567,14 @@ theorem one_eq_pi_single {i j} : (1 : Matrix n n α) i j = Pi.single (f := fun _
   simp only [one_apply, Pi.single_apply, eq_comm]
 #align matrix.one_eq_pi_single Matrix.one_eq_pi_single
 
+lemma zero_le_one_elem [Preorder α] [ZeroLEOneClass α] (i i' : n) :
+    0 ≤ (1 : Matrix n n α) i i' := by
+  by_cases hi : i = i' <;> simp [hi]
+
+lemma zero_le_one_row [Preorder α] [ZeroLEOneClass α] (i : n) :
+    0 ≤ (1 : Matrix n n α) i :=
+  fun i' => zero_le_one_elem i i'
```
Comment (Rida-Hamadani) [inline Mathlib/Data/Matrix/Basic.lean:576]:
```suggestion
  zero_le_one_elem _
```

Comment (madvorak) [reply Mathlib/Data/Matrix/Basic.lean]: Good catch, thanks!

**Comment by madvorak** [comment]
> Does it make sense to add the `ZeroLEOneClass` instance to matrix ring?

I don't know. Would it allow me to say `0 ≤ 1` also for matrices, not just for their elements and their rows?

**Comment by riccardobrasca** [comment]
Let's try and bench the result, it those are OK it cannot hurt.

**Comment by jcommelin** [comment]
Is there an ordering defined on matrices?

**Comment by madvorak** [comment]
> Is there an ordering defined on matrices?

Not to my knowledge.
If we want to define it, perhaps a separate PR?

**Comment by riccardobrasca** [comment]
I just realized that I misread the statement. Forget about by comment.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**